lib/cutting_edge/workers/dependency.rb

```diff
@@ -28,14 +29,66 @@ def perform(identifier, lang, locations, dependency_types, to_addr, auth_token =
       add_to_store(identifier, dependencies) unless @nothing_changed
     ensure
       unless @nothing_changed
+      #  alt_diff(dependencies[:locations], old_dependencies[:locations])
         badge_worker(identifier)
-        mail_worker(identifier, to_addr) if to_addr
+        mail_worker(identifier, to_addr, alt_diff(old_dependencies[:locations], dependencies[:locations])) if to_addr && !old_dependencies.empty?
```

Comment:
I wonder if the diff should be created by the worker instead, since it's reasonably computation heavy.